frontend/app/routes/public/apply/$id/terms-and-conditions.tsx

```diff
@@ -35,20 +50,78 @@ export async function loader({ context: { appContainer, session }, request, para
 
 export async function action({ context: { appContainer, session }, request, params }: Route.ActionArgs) {
   const formData = await request.formData();
+  const t = await getFixedT(request, handle.i18nNamespaces);
 
   const securityHandler = appContainer.get(TYPES.routes.security.SecurityHandler);
   securityHandler.validateCsrfToken({ formData, session });
 
+  const doNotConsent = formData.get('doNotConsent') ?? '';
+
+  if (doNotConsent) {
+    return redirect(t('apply:terms-and-conditions.dialog.exit-btn-link'));
+  }
```

Comment:
We should clear the state as well before exiting:

```suggestion
  if (doNotConsent) {
    clearApplyState({ params, session });
    return redirect(t('apply:terms-and-conditions.dialog.exit-btn-link'));
  }
```